tests/test_git_operations.py

```diff
@@ -0,0 +1,60 @@
+from unittest import mock
+
+from git_machete.git_operations import FullCommitHash, GitContext, LocalBranchShortName
+from .mockers import (get_current_commit_hash, GitRepositorySandbox, mock_run_cmd)
+
+
+class TestGitOperations:
+
+    def setup_method(self) -> None:
+
+        self.repo_sandbox = GitRepositorySandbox()
+
+        (
+            self.repo_sandbox
+            # Create the remote and sandbox repos, chdir into sandbox repo
+            .new_repo(self.repo_sandbox.remote_path, "--bare")
+            .new_repo(self.repo_sandbox.local_path)
+            .execute(f"git remote add origin {self.repo_sandbox.remote_path}")
+            .execute('git config user.email "tester@test.com"')
+            .execute('git config user.name "Tester Test"')
+        )
+
+    @mock.patch('git_machete.utils.run_cmd', mock_run_cmd)
+    def test_run_git(self) -> None:
+        """
+        Verify behaviour of a 'GitContext._run_git()' method
+        """
+        (
+            self.repo_sandbox.new_branch("master")
+                .commit("master first commit")
+        )
+        master_branch_first_commit_hash = get_current_commit_hash()
+
+        git = GitContext()
+        assert git._run_git("rev-parse", "--verify", "--quiet", master_branch_first_commit_hash + "^{commit}", allow_non_zero=True) == 0
+        assert git._run_git("rev-parse", "HEAD") == 0
+
+    def test_popen_git(self) -> None:
+        """
+        Verify behaviour of a 'GitContext._popen_git()' method
+        """
+        (
+            self.repo_sandbox.new_branch("master")
+                .commit("master first commit")
+        )
+        master_branch_first_commit_hash = get_current_commit_hash()
+        (
+            self.repo_sandbox.new_branch("develop")
+                .commit("develop commit")
+                .new_branch("feature")
+                .commit("feature commit")
+        )
+        git = GitContext()
```

Comment:
Maybe move to separate test methods, like `test_is_ancestor_or_equal`

Reply:
Each method call in that test case tests the same `_popen_git()` method and that's why i put them together

Reply:
Ok, up to you then